django/db/models/base.py

```diff
@@ -450,6 +450,14 @@ def __init__(self, db=None):
         # Necessary for correct validation of new instances of objects with explicit (non-auto) PKs.
         # This impacts validation only; it has no effect on the actual save.
         self.adding = True
+        self.cache = {}
+
+    def set_fields_cache(self, field, value):
```

Comment:
What is the benefit of having these `set_fields_cache` and `get_fields_cache` methods, over just accessing `instance._state.cache` directly? The only apparent reason is so that you can pass in a field and have `field.get_cache_name()` automatically called, but that could just as well be handled in the field mixin (and it's better handled there, as that keeps the encapsulations cleaner).